gridcapa-cse-valid-app/src/main/java/com/farao_community/farao/cse_valid/app/rao/CseValidRaoValidator.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ *  This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.farao_community.farao.cse_valid.app.rao;
+
+import com.farao_community.farao.cse_valid.api.resource.CseValidRequest;
+import com.farao_community.farao.cse_valid.api.resource.ProcessType;
+import com.farao_community.farao.cse_valid.app.FileExporter;
+import com.farao_community.farao.cse_valid.app.FileImporter;
+import com.farao_community.farao.data.crac_api.Crac;
+import com.farao_community.farao.data.crac_creation.creator.cse.CseCrac;
+import com.farao_community.farao.data.rao_result_api.OptimizationState;
+import com.farao_community.farao.data.rao_result_api.RaoResult;
+import com.farao_community.farao.rao_runner.api.resource.RaoRequest;
+import com.farao_community.farao.rao_runner.api.resource.RaoResponse;
+import com.farao_community.farao.rao_runner.starter.RaoRunnerClient;
+import com.powsybl.iidm.network.Network;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.time.OffsetDateTime;
+
+/**
+ * @author Oualid Aloui {@literal <oualid.aloui at rte-france.com>}
+ */
+
+@Component
+public class CseValidRaoValidator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CseValidRaoValidator.class);
+
+    private final FileImporter fileImporter;
+    private final FileExporter fileExporter;
+    private final RaoRunnerClient raoRunnerClient;
+    private int variantCounter = 0;
```

Comment:
I generally don't like instance attributes in a Component as it may cause problems in case of parallel use of the bean.
What do you use the `variantCounter` for here?

Reply:
Variant counter is used in the RAO validator because it is inside the dichotomy so it counts the iteration number of the dichotomy in the end, but it is not a @Component object it's a final class :smiley: So indeed it is not necessary here.